src/library/libraryfeature.cpp

```diff
@@ -36,3 +47,49 @@ QStringList LibraryFeature::getPlaylistFiles(QFileDialog::FileMode mode) const {
     if (! dialog.exec()) return QStringList();
     return dialog.selectedFiles();
 }
+
+bool LibraryFeature::exportPlaylistItemsIntoFile(
+        QString playlistFilePath,
+        const QList<QString>& playlistItemLocations,
+        bool useRelativePath)    {
+    if (playlistFilePath.endsWith(
+            QStringLiteral(".pls"),
+            Qt::CaseInsensitive)) {
+        return ParserPls::writePLSFile(
+                playlistFilePath,
+                playlistItemLocations,
+                useRelativePath);
+    } else if (playlistFilePath.endsWith(
+            QStringLiteral(".m3u8"),
+            Qt::CaseInsensitive)) {
+        return ParserM3u::writeM3U8File(
+                playlistFilePath,
+                playlistItemLocations,
+                useRelativePath);
+    } else {
+        //default export to M3U if file extension is missing
+        if (!playlistFilePath.endsWith(
+                QStringLiteral(".m3u"),
+                Qt::CaseInsensitive)) {
+            kLogger.debug()
+                    << "No valid file extension for playlist export specified."
+                    << "Appending .m3u and exporting to M3U.";
+            playlistFilePath.append(QStringLiteral(".m3u"));
+            if (QFileInfo(playlistFilePath).exists()) {
+                auto overwrite = QMessageBox::question(
+                        NULL,
+                        tr("Overwrite File?"),
+                        tr("Since you have specified no extension, m3u is used, but the file \"%1"
```

Comment:
@Be-ing How about the punctuation? I would prefer to use simple, short sentences. One statement followed by a question.

"Overwrite Playlist File?"
"The playlist file \"%1\" already exists. Do you really want to overwrite it?"

The user does not need to know how the file name is composed. Only what the resulting file name is and that there is a conflict, namely that the file already exists and will be overwritten when continuing.

Reply:
The problem is that the user might expect the filter he selected, such as PLS, to determine the format. So I wanted to explicitly point to this to avoid confusion. This is a very edge case and I'd prefer to nudge the user into properly specifying a format with this dialog.

Reply:
@uklotzde would you omit that information regardless?

Reply:
How about:
"Overwrite Playlist File?"
"The playlist file \"%1\" already exists (\"m3u\" extension was added because you did not pick an extension). Do you really want to overwrite it?"

Reply:
Let's remove the braces and split it into two sentences:

A playlist file with the name "%1" already exists. The default ".m3u" extension was added, because you did not specify/pick/select one.

Do you really want to overwrite it?

The final question should be separated from the explanation to emphasize it.

Reply:
Not sure about the verb, just choose what applies.

Reply:
Yes, that's better. Just remove the comma after "added".

Reply:
That seems sensible - how about this then:
```
A playlist file with the name "%1" already exists. 
The default ".m3u" extension was added because none was specified. 
```
I deliberately used passive, since the extension will usually be added automatically, the user adding it manually is the exception.

Although, with these 2 sentences, they aren't obviously linked. It somehow feels weird to me, unlike the version with parenthesis.

How about this?
```
Using the default ".m3u" extension because none was specified.
Do you want to really want to overwrite "%1"?
```

Reply:
I prefer 

> A playlist file with the name "%1" already exists. 
The default ".m3u" extension was added because none was specified.

Reply:
ping

Reply:
Sorry, have a lot going on as my university semester is wrapping up. I still think that these two sentences don't feel properly connected, but lets roll with it - as I said its a rare edge case anyways ^^

Reply:
By the way, I previously used "Do you wish" for the prompt since that is used for the default Qt overwrite dialog:

![image](https://user-images.githubusercontent.com/13354331/79645249-6b237580-81ae-11ea-9978-af9b9384c41f.png)

Reply:
I have updated and tested the dialog - let me know if we should adjust it, otherwise it's ready for merge :)

Reply:
Looks good to me, thanks